adapters/amqp-vertx/src/test/java/org/eclipse/hono/adapter/amqp/impl/VertxBasedAmqpProtocolAdapterTest.java

```diff
@@ -851,6 +851,22 @@ public void testMessageLimitExceededForAnEventMessage(final TestContext ctx) {
                 }));
     }
 
+    private String getCommandEndpoint() {
+        return useLegacyCommandEndpoint() ? CommandConstants.COMMAND_LEGACY_ENDPOINT : CommandConstants.COMMAND_ENDPOINT;
+    }
+
+    /**
+     * Checks whether the legacy Command & Control endpoint shall be used.
+     * <p>
+     * Returns {@code false} by default. Subclasses may return {@code true} here to perform tests using the legacy
+     * command endpoint.
+     *
+     * @return {@code true} if the legacy command endpoint shall be used.
+     */
+    protected boolean useLegacyCommandEndpoint() {
```

Comment:
what about using [parameterized tests](https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests) for running the tests with the legacy and the new endpoint name instead of using a subclass and introducing this method?

Reply:
I had a look into that but since there is no parameterized test definition at class level in JUnit 5 yet, I decided against it (I didn't want to annotate each method).
When we remove support for the legacy endpoints, we can just delete the subclasses.

Reply:
fair enough